src/charm.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Retrieves certificates from an ACME server using the aws route53 dns provider."""
+
+import logging
+from typing import Dict
+
+from charms.acme_client_operator.v0.acme_client import AcmeClient  # type: ignore[import]
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+
+class Route53LegoOperatorCharm(AcmeClient):
+    """Main class that is instantiated every time an event occurs."""
+
+    def __init__(self, *args):
+        """Uses the acme_client library to manage events."""
+        super().__init__(*args, plugin="route53")
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    @property
+    def _aws_access_key_id(self):
+        """Returns aws access key from config."""
+        return self.model.config.get("aws_access_key_id")
+
+    @property
+    def _aws_hosted_zone_id(self):
+        """Returns aws hosted zone id from config."""
+        return self.model.config.get("aws_hosted_zone_id")
+
+    @property
+    def _aws_secret_access_key(self):
+        """Returns aws secret access key from config."""
+        return self.model.config.get("aws_secret_access_key")
+
+    @property
+    def _aws_region(self):
+        """Returns aws region from config."""
+        return self.model.config.get("aws_region")
+
+    @property
+    def _aws_shared_credentials_file(self):
+        """Returns aws shared credentials file path from config."""
+        return self.model.config.get("aws_shared_credentials_file")
```

Comment:
As above, I think we should get rid of this.

Reply:
✅